src/modules.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+use crate::ansi;
+use crate::deno_dir::DenoDir;
+use crate::libdeno::deno_mod;
+use crate::msg;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::fmt;
+
+pub struct ModuleInfo {
+  name: String,
+  children: Vec<deno_mod>,
+}
+
+/// A collection of JS modules.
+pub struct Modules {
+  pub info: HashMap<deno_mod, ModuleInfo>,
+  pub by_name: HashMap<String, deno_mod>,
+}
+
+impl Modules {
+  pub fn new() -> Modules {
+    Self {
+      info: HashMap::new(),
+      by_name: HashMap::new(),
+    }
+  }
+
+  pub fn get_id(&self, name: &str) -> Option<deno_mod> {
+    self.by_name.get(name).map(|id_ref| *id_ref)
```

Comment:
You are using an explicit closure for cloning elements
note: #[warn(clippy::map_clone)] on by default
help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_clone
help: Consider calling the dedicated `cloned` method: `self.by_name.get(name).cloned()`

Reply:
Fixed.